### PR TITLE
creating webapp library

### PR DESCRIPTION
**Pull request by deontpearson**
# Description
This PR is to further abstract and simplify usage of the chart library. The library now defines the `asimmetric` helms charts that can be used in every service. The library enables the deployment of 3 kubernetes services.

- Webapps: will create k8s deployments + services with the correct ports and health probes
- Workers: will create k8s deployments with no service or port. These are used for long running jobs like consumers / producers
- Cronjobs: will create k8s cronjobs which creates jobs at a specified time interval

History on the progress of the chart-library can be found at https://github.com/Asimmetric/onboarding/wiki/Chart-Library

The `charts/asimmetric/README.md` file in this PR should be used as a reference when defining helm values, as well as a source of information for how the library is used

# Suggested Resources
- what is helm (https://helm.sh/docs/intro/quickstart/)
- what are helm templates (https://helm.sh/docs/chart_template_guide/getting_started/)
- what are helm value files (https://helm.sh/docs/chart_template_guide/values_files/)

### CODE REVIEW

File: charts/webapp-example/values.yaml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+cronjobs:
+- name: cronjob-1
+- name: cronjob-2
+
+workers:
+- name: worker-1
+- name: worker-2
+
+webapp:
```
Comment (brandon-piner-aruba) [inline charts/webapp-example/values.yaml:21]: webapps plural

File: charts/webapp-example/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v2
+name: webapp-example
+description: A Helm chart for Asimmetric
+type: application
+version: 1.0.0
+dependencies:
+- name: webapp
```
Comment (brandon-piner-aruba) [inline charts/webapp-example/Chart.yaml:7]: asimmetric or...

File: charts/webapp-example/values.yaml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      # name: webapp-example
+      # annotations: {}
+    # overrides the command from the dockerfile
+    command: ["printenv"]
+    # the container port to expose
+    port: 8000
+    # customize the service created
+    # TODO: add external IP service account
+    # TODO: add external name service account
+    service: {}
```
Comment (brandon-piner-aruba) [inline charts/webapp-example/values.yaml:44]: redundant to define service here

File: charts/asimmetric-example/output.yaml [hunk 4]
```diff
@@ -1,3 +1,10 @@
+Hang tight while we grab the latest from your chart repositories...
```
Comment (brandon-piner-aruba) [inline charts/asimmetric-example/output.yaml:1]: The output yaml should not have this text. There is a little change to the command you used to create this

**Review by brandon-piner-aruba — COMMENTED** [review]
Looks great!

1) update the .github workflow
2) delete the old libs that we wont be using. 
3) a document on more or less how to convert from the old lib to the new. Keep it basic because your actual docs in the example are good

File: examples/asimmetric-example/values-staging.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    envSealedSecrets:
+      aws-access:
+        AWS_REGION: sealed_version_of_the_base64_encoded_aws_region
+        AWS_ACCESS_ID: sealed_version_of_the_base64_encoded_aws_access_id
+  applications:
+    example-service:
+      role: webapp
+      replicaCount: 2
+      image:
+        tag: "example-service-tag"
```
Comment (brandon-piner-aruba) [inline examples/asimmetric-example/values-staging.yaml:23]: Is there a reason why this needs to be in quotes?

**Review by brandon-piner-aruba — APPROVED** [review]

File: examples/asimmetric-example/values-staging.yaml (hunk 5 above)
Comment (deontpearson) [reply examples/asimmetric-example/values-staging.yaml:23]: no reason, ill fix

**Pull request merged**